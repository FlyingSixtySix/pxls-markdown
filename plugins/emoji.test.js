const { processor, expectProcessorParse } = require('../testUtil.js');
const emoji = require('./emoji.js');

const emojiDB = { dog: '🐶' };
const emojiRegex = /🐶/i;

test('crashes when the processor isn\'t configured correctly', () => {
	// Freeze processors so that plugins run.
	expect(() => processor().use(emoji).freeze()).toThrow();
	expect(() => processor().use(emoji, { emojiDB }).freeze()).toThrow();
	expect(() => processor().use(emoji, { emojiRegex }).freeze()).toThrow();
})

const configuredProcessor = processor().use(emoji, {
	emojiDB,
	emojiRegex
});

test('tokenizes ":dog:"', () => expectProcessorParse(':dog:', false, configuredProcessor).toMatchSnapshot());
test('tokenizes "🐶"', () => expectProcessorParse('🐶', false, configuredProcessor).toMatchSnapshot());

