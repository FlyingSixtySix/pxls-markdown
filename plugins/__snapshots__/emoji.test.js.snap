// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenizes ":dog:" 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "emojiName": "dog",
        "type": "emoji",
        "value": "🐶",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`tokenizes "🐶" 1`] = `
Array [
  Object {
    "children": Array [
      Object {
        "emojiName": "dog",
        "type": "emoji",
        "value": "🐶",
      },
    ],
    "type": "paragraph",
  },
]
`;
